init-distro-rid.sh

```diff
@@ -0,0 +1,176 @@
+        if [ -e "${crossBuildLocation}/etc/redhat-release" ]; then
+            local redhatRelease=$(<${crossBuildLocation}/etc/redhat-release)
+
+            if [[ "${redhatRelease}" == "CentOS release 6."* || "$redhatRelease" == "Red Hat Enterprise Linux Server release 6."* ]]; then
+                nonPortableBuildID="rhel.6-${__BuildArch}"
+            fi
+
+        elif [ -e /etc/os-release ]; then
+            source "${crossBuildLocation}/etc/os-release"
+            if [ "${ID}" = "rhel" ]; then
+                # RHEL should have been caught by the /etc/redhat-release
+                echo "Error, please verify that your install of RedHat includes"
+                echo "/etd/redhat-release"
+                exit 1
+            fi
+
+            if [ "${ID}" = "alpine" ]; then
+                nonPortableBuildID="linux-musl-${__BuildArch}"
+            fi
+
+        elif [ -e $ROOTFS_DIR/android_platform ]; then
+            source $ROOTFS_DIR/android_platform
+            nonPortableBuildID="$RID"
+        fi
+    fi
+
+    if [ "$__BuildOS" = "FreeBSD" ]; then
+        __freebsd_version=`sysctl -n kern.osrelease | cut -f1 -d'.'`
+        nonPortableBuildID="freebsd.$__freebsd_version-$__Arch"
+    fi
+
+    if [ "${nonPortableBuildID}" != "" ]; then
+        export __DistroRid=${nonPortableBuildID}
+        export __PortableBuild=0
+    fi
+}
+
+
+# initDistroRidGlobal
+#
+# Input:
+#   os (str)
+#   arch (str)
+#   ROOTFS_DIR? (nullable vararg:string)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# The following global state is modified:
+#
+#   __BuildOS
+#   __BuildArch
+#   ROOTFS_DIR
+#
+# The following out parameters are returned
+#
+#   __DistroRid
+#   __PortableBuild
+#
+initDistroRidGlobal()
+{
+    # __DistroRid must be set at the end of the function.
+    # Previously we would create a variable __HostDistroRid and/or __DistroRid.
+    #
+    # __HostDistroRid was used in the case of a non-portable build, it has been
+    # deprecated. Now only __DistroRid is supported. It will be used for both
+    # portable and non-portable rids and will be used in build-packages.sh
+
+    export __BuildOS=$1
+    export __BuildArch=$2
+    export ROOTFS_DIR=$3
+
+    # Setup whether this is a crossbuild. We can find this out if ROOTFS_DIR
+    # is set. 
+    local isCrossBuild=0
+
+    if [ -z "${ROOTFS_DIR}" ]; then
+        isCrossBuild=0
+    else
+        # We may have a cross build. Check for the existance of the ROOTFS_DIR
+        if [ -e ${ROOTFS_DIR} ]; then
+            isCrossBuild=1
+        else
+            echo "Error ROOTFS_DIR has been passed, but the location is not valid."
+            exit 1
+        fi
+    fi
+
+    if (( ${isCrossBuild} == 1 )); then
+        initNonPortableDistroRid ${isCrossBuild}
+    else
+        initNonPortableDistroRid
+    fi
+
+    if [ -z "${__DistroRid}" ]; then
+        # The non-portable build rid was not set. Set the portable rid.
+
+        export __PortableBuild=1
```

Comment:
I don't think we should export this.

Reply:
As per our conversation the script will now set portable build. This will no longer be allowed to be overriden. There is no current valid case that allowing enables.